packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -243,6 +248,82 @@ export class CdkToolkit {
     }
   }
 
+  public async watch(options: WatchOptions) {
+    const rootDir = path.dirname(path.resolve(PROJECT_CONFIG));
+    debug("root directory used for 'watch' is: %s", rootDir);
+
+    const watchSettings: { include?: string | string[], exclude: string | string [] } | undefined =
+        this.props.configuration.settings.get(['watch']);
+    if (!watchSettings) {
+      throw new Error("Cannot use the 'watch' command without specifying at least one directory to monitor. " +
+        'Make sure to add a "watch" key to your cdk.json');
+    }
+
+    // For the "include" subkey under the "watch" key, the behavior is:
+    // 1. No "watch" setting? We error out.
+    // 2. "watch" setting without an "include" key? We default to observing "./**".
+    // 3. "watch" setting with an empty "include" key? We default to observing "./**".
+    // 4. Non-empty "include" key? Just use the "include" key.
+    const watchIncludes = this.patternsArrayForWatch(watchSettings.include, rootDir, /* returnRootDirIfEmpty */ true);
+    debug("'include' patterns for 'watch': %s", watchIncludes);
+
+    // For the "exclude" subkey under the "watch" key,
+    // the behavior is to add some default excludes in addition to the ones specified by the user:
+    // 1. The CDK output directory.
+    // 2. Any file whose name starts with a dot.
+    // 3. Any directory's content whose name starts with a dot.
+    // 4. Any node_modules and its content.
+    const outputDir = this.props.configuration.settings.get(['output']);
+    const watchExcludes = this.patternsArrayForWatch(watchSettings.exclude, rootDir, /* returnRootDirIfEmpty */ false).concat(
+      `${rootDir}/${outputDir}/**`,
```

Comment:
I think the cdk.out directory should _always_ implicitly be added (even if there's user-defined `exclude`). The rest of the defaults should be only added if `exclude` is not specified.

Reply:
Really? You don't think we should always ignore the files and directories starting with `.`, like `.git`, and Vim backup files, for example?

Reply:
Maybe I misunderstood the code. Is this exclude list always added to the user defined exclude list or is it just used if there is no user defined list?

If this list is always excluded, then I am fine with it. But then users should be able to negate specific files manually

Reply:
These 4 excludes (CDK output directory, files starting with `.`, directories starting with `.`, and `node_modules`) are always added, regardless whether the user specified the `"exclude"` key in their `cdk.json`.

Reply:
> If this list is always excluded, then I am fine with it. But then users should be able to negate specific files manually

By this, are you saying they should be able to negate our exclusion, and so observe a file that we exclude by default (for example, they want to observe a given directory contents whose name starts with a `.`)?

Reply:
Yes

Reply:
@eladb turning off the default excludes is turning out to be more complicated than I initially thought. Basically, the way negative patterns are used in `chokidar` means "everything except what's matched by this pattern". So, there is no point in using them in the `"exclude"` subkey of `cdk.json`, as, as [this comment](https://github.com/paulmillr/chokidar/issues/58#issuecomment-106283112) from a chokidar issue states, it's a double negative there.

The way chokidar works is basically: "a file is matched if it a) is included in one of the `include` paths, and b) is _not_ included in any `exclude` paths". This way of working, while sensible, means though that there's no way of "turning off" a default `exclude` like we add in this PR - the files delimited by the default excludes will always be excluded.

Now, there are a few ways we can handle this:
1. Do nothing - just not allow customers to override the default excludes.
2. Stop defaulting excludes - customers will have to specify them theirselves (our init template changes might help here).
3. Add some logic to look for `"exclude"` patterns that start with a `!`, and remove all exclude patterns that match that pattern after the bang - so, `"!**/.*"` would remove the default exclude `"**/.*"`. Since chokidar doesn't use negated patterns, there's no risk in using them for our own purposes. We would have to document this, of course.
4. Add another subkey under the `"watch"` key, something like "useDefaultExcludes"`, that will be `true` by default, but a customer can set it to `false` to get rid of the default excludes. Again, we would have to document this properly.

Any thoughts on which one would you prefer?